### ISSUE TITLE
`Concatenate` false negative (caught by `pyright`, but not `mypy`)

### ISSUE THREAD

**Issue by supersergiy** [python/mypy#14169]
**Bug Report**
False negative in the following example:

**To Reproduce**
[Playground Link](https://mypy-play.net/?mypy=latest&python=3.11&gist=8975ccee3891aa457f880c144a890810)
```python
from typing import Protocol
from typing_extensions import ParamSpec, Concatenate

P = ParamSpec("P")

class A(Protocol[P]):
    def foo(self, *args: P.args, **kwargs: P.kwargs):
        ...
    
def bar(b: A[P]) -> A[Concatenate[int, P]]:
    return b # no error
```

**Expected Behavior**

Type error on `return b`, which is what pyright does ([pyright playground](https://pyright-playground.decorator-factory.su/?gzip=H4sIAOGFfWMC_01O0QrCMAx871cEn7ox-wEDheEPFHwcQ7rajuLWjLSi_r2tc86DkEsud8QSThBfs_MDuGlGiiAJI2ocmd20i3lG44NDH35nitR0no2u4IReq6SnYkzCYdP4Tu4KxvSoQoCGr9Gt7IqaQcLVWLCIPJjRVlAqGkINUuSexvL2WDcL-7oyhBAfznJEr4j3NTQ5GPbHRP5-ap2PFciuW8xk4p089PAGjjDmzvwAAAA%3D))

**Actual Behavior**

No type error.

**Your Environment**

- Mypy version used: 0.991
